File: app/src/main/java/com/example/mike/mp3player/client/views/buttons/SkipToPreviousButton.java
package com.example.mike.mp3player.client.views.buttons;

import android.content.Context;
import android.os.Handler;
import android.view.View;
import android.widget.ImageView;

import com.example.mike.mp3player.R;
import com.example.mike.mp3player.client.MediaControllerAdapter;

import javax.inject.Inject;
import javax.inject.Named;

public class SkipToPreviousButton extends MediaButton {

    @Inject
    public SkipToPreviousButton(Context context, MediaControllerAdapter mediaControllerAdapter, @Named("main") Handler mainUpdater) {
        super(context, mediaControllerAdapter, mainUpdater);
    }

    @Override
    public void init(ImageView view) {
        super.init(view);
        setImage(R.drawable.ic_baseline_skip_previous_24px);
    }

    @Override
    public void onClick(View view) {
        mediaControllerAdapter.skipToPrevious();
    }

}

File: client/src/main/java/com/github/goldy1992/mp3player/client/views/buttons/SkipToPreviousButton.kt
package com.github.goldy1992.mp3player.client.views.buttons

import android.content.Context
import android.view.View
import com.github.goldy1992.mp3player.client.MediaControllerAdapter
import com.github.goldy1992.mp3player.R
import com.google.android.material.button.MaterialButton
import dagger.hilt.android.qualifiers.ActivityContext
import dagger.hilt.android.scopes.FragmentScoped
import javax.inject.Inject

@FragmentScoped
class SkipToPreviousButton @Inject constructor(@ActivityContext context: Context,
                                               mediaControllerAdapter: MediaControllerAdapter)
    : MediaButton(context, mediaControllerAdapter) {

    override fun init(imageView: MaterialButton) {
        super.init(imageView)
        setImage(R.drawable.ic_baseline_skip_previous_24px)
    }

    override fun onClick(view: View?) {
        mediaControllerAdapter.skipToPrevious()
    }

    override fun logTag(): String {
        return "SKIP_PRV_BTN"
    }
}